src/hammer-vlsi/technology/sky130/__init__.py

```diff
@@ -66,60 +68,95 @@ def setup_verilog(self) -> None:
+                for line in sf:
+                    line = line.replace('`default_nettype none','`default_nettype wire')
+                    df.write(line)
 
     # Copy and hack the tech-lef, adding this very important `licon` section
     def setup_techlef(self) -> None:
         setting_dir = self.get_setting("technology.sky130.sky130A")
         setting_dir = Path(setting_dir)
-        tlef_old_path = setting_dir / 'libs.ref' / self.library_name / 'techlef' / f'{self.library_name}.tlef'
-        if not tlef_old_path.exists():
-            raise FileNotFoundError(f"Tech-LEF not found: {tlef_old_path}")
+        source_path = setting_dir / 'libs.ref' / self.library_name / 'techlef' / f'{self.library_name}__nom.tlef'
+        if not source_path.exists():
+            raise FileNotFoundError(f"Tech-LEF not found: {source_path}")
 
         cache_tech_dir_path = Path(self.cache_dir)
         os.makedirs(cache_tech_dir_path, exist_ok=True)
-        tlef_new_path = cache_tech_dir_path / f'{self.library_name}.tlef'
+        dest_path = cache_tech_dir_path / f'{self.library_name}__nom.tlef'
+
+        with open(source_path, 'r') as sf:
+            with open(dest_path, 'w') as df:
+                self.logger.info("Modifying Technology LEF: {} -> {}".format
+                    (source_path, dest_path))
+                for line in sf:
+                    df.write(line)
+                    if line.strip() == 'END pwell':
+                        df.write(_the_tlef_edit)
+
+    # Copy and hack the klayout techfile, to add all required LEFs
+    # TODO: should this be in OpenROAD plugin instead?
+    def setup_klayout_techfile(self) -> None:
+        setting_dir = self.get_setting("technology.sky130.sky130A")
+        setting_dir = Path(setting_dir)
 
-        f_old = open(tlef_old_path,'r')
-        f_new = open(tlef_new_path,'w')
-        for line in f_old:
-            f_new.write(line)
-            if line.strip() == 'END pwell':
-                f_new.write(_the_tlef_edit)
-        f_old.close()
-        f_new.close()
+        source_path = setting_dir / 'libs.tech' / 'klayout' / 'sky130A.lyt'
+        if not source_path.exists():
+            raise FileNotFoundError(f"Klayout techfile not found: {source_path}")
+
+        cache_tech_dir_path = Path(self.cache_dir)
+        os.makedirs(cache_tech_dir_path, exist_ok=True)
+        dest_path = cache_tech_dir_path / f'sky130A.lyt'
+
+        insert_lines=''
+        lef_files = self.read_libs([
+            hammer_tech.filters.lef_filter
+        ], hammer_tech.HammerTechnologyUtils.to_plain_item)
+
+        for i,lef_file in enumerate(lef_files):
+            # skip standard cell lef file (klayout gives an error that cells already exist)
+            # TODO: very hacky to assume std cell lef is in position 1, is there a way to only extract tech lef + macro lefs?
+            if i == 1: continue 
+            insert_lines += f"<lef-files>{lef_file}</lef-files>\n"
+
```

Comment:
Is there a clean way to only extract the tech lef + macro lefs?

Reply:
Do something similar to this: https://github.com/ucb-bar/hammer-cadence-plugins/blob/master/power/voltus/__init__.py#L112-L124